rpc/src/eth_rpc.rs

```diff
@@ -210,27 +211,36 @@ impl EthApi {
         let body = self.fetch_block_body(header.hash()).await?;
         let transactions =
             BlockTransactions::Hashes(body.transactions().iter().map(Transaction::hash).collect());
-        let uncles = body
-            .uncles()
-            .unwrap_or_default()
-            .iter()
-            .map(|uncle| uncle.hash())
-            .collect();
+        let uncles = body.uncles().iter().map(|uncle| uncle.hash()).collect();
         let withdrawals = body
             .withdrawals()
             .map(|withdrawals| withdrawals.iter().map(Withdrawal::from).collect());
 
-        // TODO: Add calculation for the block's size:
+        // Calculate block size:
         //   len(rlp(header, transactions, uncles, withdrawals))
-        // NOTE: Transactions should be encoded with envelope
-        let size = None;
+        // Note: transactions are encoded with header
+        let size = {
+            let payload_size = header.length()
```

Comment:
```suggestion
            let payload_size = header.length()
```

Our Header type here doesn't use the macro or implement length in its implementation of `Encodable` so we are stilling encoding to find the length of `Header`. Would you want to implement `length()` for header in this PR or make an issue for it any we can solve it in another PR.


This would be the only place in calculating size where we still are encoding to get the length I believe.

Reply:
I will do it in a separate PR (no need to open issue about it, I will do it right now).